stress/shell/src/main/java/alluxio/stress/cli/GetPinnedFileIdsBench.java

```diff
@@ -0,0 +1,136 @@
+  private GetPinnedFileIdsParameters mParameters = new GetPinnedFileIdsParameters();
+
+  private final InstancedConfiguration mConf = InstancedConfiguration.defaults();
+  private final FileSystemContext mFileSystemContext;
+  private final AlluxioURI mBaseUri;
+
+  {
+    mFileSystemContext = FileSystemContext.create(mConf);
+    mBaseUri = new AlluxioURI(mParameters.mBasePath);
+  }
+
+  @Override
+  public void prepare() throws Exception {
+    try (CloseableResource<alluxio.client.file.FileSystemMasterClient> client =
+             mFileSystemContext.acquireMasterClientResource()) {
+      client.get().createDirectory(mBaseUri,
+              CreateDirectoryPOptions.newBuilder().setAllowExists(true).build());
+    }
+    int numFiles = mParameters.mNumFiles;
+    int fileNameLength = (int) Math.max(8, Math.log10(numFiles));
+
+    CompletableFuture<Void>[] futures = new CompletableFuture[numFiles];
+    LOG.info("Generating {} test files", numFiles);
+    for (int i = 0; i < numFiles; i++) {
+      AlluxioURI fileUri = mBaseUri.join(CommonUtils.randomAlphaNumString(fileNameLength));
+      CompletableFuture<Void> future = CompletableFuture.supplyAsync((Supplier<Void>) () -> {
+        try (CloseableResource<FileSystemMasterClient> client =
+                 mFileSystemContext.acquireMasterClientResource()) {
+          client.get().createFile(fileUri,
+              CreateFilePOptions
+                  .newBuilder()
+                  .setBlockSizeBytes(mConf.getBytes(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT))
+                  .build());
+          client.get().setAttribute(fileUri,
+              SetAttributePOptions
+                  .newBuilder()
+                  .setPinned(true)
+                  .build());
+        } catch (AlluxioStatusException e) {
+          LOG.warn("Exception during file creation of {}", fileUri, e);
+        }
+        return null;
+      }, getPool());
+      futures[i] = (future);
+    }
+    CompletableFuture.allOf(futures).join();
+  }
+
+  @Override
+  public void cleanup() throws Exception {
+    try (CloseableResource<alluxio.client.file.FileSystemMasterClient> client =
+             mFileSystemContext.acquireMasterClientResource()) {
+      client.get().delete(mBaseUri, DeletePOptions.newBuilder().setRecursive(true).build());
+    }
+    super.cleanup();
+  }
+
+  @Override
+  public RpcTaskResult runRPC() throws Exception {
+    PinListFileSystemMasterClient client = new PinListFileSystemMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build());
+    RpcTaskResult result = new RpcTaskResult();
+    result.setBaseParameters(mBaseParameters);
+    result.setParameters(mParameters);
+
+    long durationMs = FormatUtils.parseTimeSize(mParameters.mDuration);
+    Stopwatch durationStopwatch = Stopwatch.createStarted();
+    Stopwatch pointStopwatch = Stopwatch.createUnstarted();
+
+    while (durationStopwatch.elapsed(TimeUnit.MILLISECONDS) < durationMs) {
```

Comment:
what does this test do?

Reply:
`durationMs` is the time how long the benchmark should run, we check the running time against this to know when we should stop the benchmark.